lib/msf/core/post/file.rb

```diff
@@ -181,6 +181,8 @@ def expand_path(path)
   #
   # @param path [String] Remote filename to check
   def file?(path)
+    return false if path.nil?
```

Comment:
`rm_rf` was being called in the module with nil: `rm_f(lua_file, exe_file)`. This was previously safe to do, but was broken by this change from `exist?` to `file?`

https://github.com/rapid7/metasploit-framework/pull/15403/files#diff-cc6585bd12e8d43e126c2e1a79e4132f56c1e180483f410ccbf90cfb04c5e5c8R478